<?xml version="1.0" encoding="ascii"?>
<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Transitional//EN"
          "DTD/xhtml1-transitional.dtd">
<html xmlns="http://www.w3.org/1999/xhtml" xml:lang="en" lang="en">
<head>
  <title>script-phyloFixedVar</title>
  <link rel="stylesheet" href="epydoc.css" type="text/css" />
  <script type="text/javascript" src="epydoc.js"></script>
</head>

<body bgcolor="white" text="black" link="blue" vlink="#204080"
      alink="#204080">
<!-- ==================== NAVIGATION BAR ==================== -->
<table class="navbar" border="0" width="100%" cellpadding="0"
       bgcolor="#a0c0ff" cellspacing="0">
  <tr valign="middle">

  <!-- Tree link -->
      <th>&nbsp;&nbsp;&nbsp;<a
        href="module-tree.html">Trees</a>&nbsp;&nbsp;&nbsp;</th>

  <!-- Index link -->
      <th>&nbsp;&nbsp;&nbsp;<a
        href="identifier-index.html">Indices</a>&nbsp;&nbsp;&nbsp;</th>

  <!-- Help link -->
      <th>&nbsp;&nbsp;&nbsp;<a
        href="help.html">Help</a>&nbsp;&nbsp;&nbsp;</th>

      <th class="navbar" width="100%"></th>
  </tr>
</table>
<table width="100%" cellpadding="0" cellspacing="0">
  <tr valign="top">
    <td width="100%">
      <span class="breadcrumbs">
        Script&nbsp;phyloFixedVar
      </span>
    </td>
    <td>
      <table cellpadding="0" cellspacing="0">
        <!-- hide/show private -->
        <tr><td align="right"><span class="options">[<a href="javascript:void(0);" class="privatelink"
    onclick="toggle_private();">hide&nbsp;private</a>]</span></td></tr>
        <tr><td align="right"><span class="options"
            >[<a href="frames.html" target="_top">frames</a
            >]&nbsp;|&nbsp;<a href="script-phyloFixedVar-module.html"
            target="_top">no&nbsp;frames</a>]</span></td></tr>
      </table>
    </td>
  </tr>
</table>
<!-- ==================== SCRIPT DESCRIPTION ==================== -->
<h1 class="epydoc">Script phyloFixedVar</h1><p class="nomargin-top"></p>
<!-- ==================== CLASSES ==================== -->
<a name="section-Classes"></a>
<table class="summary" border="1" cellpadding="3"
       cellspacing="0" width="100%" bgcolor="white">
<tr bgcolor="#70b0f0" class="table-header">
  <td colspan="2" class="table-header">
    <table border="0" cellpadding="0" cellspacing="0" width="100%">
      <tr valign="top">
        <td align="left"><span class="table-header">Classes</span></td>
        <td align="right" valign="top"
         ><span class="options">[<a href="#section-Classes"
         class="privatelink" onclick="toggle_private();"
         >hide private</a>]</span></td>
      </tr>
    </table>
  </td>
</tr>
<tr>
    <td width="15%" align="right" valign="top" class="summary">
      <span class="summary-type">&nbsp;</span>
    </td><td class="summary">
        <a href="script-phyloFixedVar.var-class.html" class="summary-name">var</a><br />
      Create var object to stock variant information
    </td>
  </tr>
<tr>
    <td width="15%" align="right" valign="top" class="summary">
      <span class="summary-type">&nbsp;</span>
    </td><td class="summary">
        <a href="script-phyloFixedVar.genome-class.html" class="summary-name">genome</a><br />
      Create genome object to stock all of its variants informations
    </td>
  </tr>
<tr>
    <td width="15%" align="right" valign="top" class="summary">
      <span class="summary-type">&nbsp;</span>
    </td><td class="summary">
        <a href="script-phyloFixedVar.node-class.html" class="summary-name">node</a><br />
      Create node object to stock the phylogeny tree
    </td>
  </tr>
</table>
<!-- ==================== FUNCTIONS ==================== -->
<a name="section-Functions"></a>
<table class="summary" border="1" cellpadding="3"
       cellspacing="0" width="100%" bgcolor="white">
<tr bgcolor="#70b0f0" class="table-header">
  <td colspan="2" class="table-header">
    <table border="0" cellpadding="0" cellspacing="0" width="100%">
      <tr valign="top">
        <td align="left"><span class="table-header">Functions</span></td>
        <td align="right" valign="top"
         ><span class="options">[<a href="#section-Functions"
         class="privatelink" onclick="toggle_private();"
         >hide private</a>]</span></td>
      </tr>
    </table>
  </td>
</tr>
<tr>
    <td width="15%" align="right" valign="top" class="summary">
      <span class="summary-type">parser object</span>
    </td><td class="summary">
      <table width="100%" cellpadding="0" cellspacing="0" border="0">
        <tr>
          <td><span class="summary-sig"><a href="script-phyloFixedVar-module.html#get_parser" class="summary-sig-name">get_parser</a>()</span><br />
      Parse arguments</td>
          <td align="right" valign="top">
            
            
          </td>
        </tr>
      </table>
      
    </td>
  </tr>
<tr>
    <td width="15%" align="right" valign="top" class="summary">
      <span class="summary-type">&nbsp;</span>
    </td><td class="summary">
      <table width="100%" cellpadding="0" cellspacing="0" border="0">
        <tr>
          <td><span class="summary-sig"><a href="script-phyloFixedVar-module.html#printObjNucl" class="summary-sig-name">printObjNucl</a>(<span class="summary-sig-arg">objetNucl</span>)</span><br />
      Debug fonction, print all attributes of var object</td>
          <td align="right" valign="top">
            
            
          </td>
        </tr>
      </table>
      
    </td>
  </tr>
<tr>
    <td width="15%" align="right" valign="top" class="summary">
      <span class="summary-type">node object</span>
    </td><td class="summary">
      <table width="100%" cellpadding="0" cellspacing="0" border="0">
        <tr>
          <td><span class="summary-sig"><a href="script-phyloFixedVar-module.html#createNode" class="summary-sig-name">createNode</a>()</span><br />
      Create a new empty node object</td>
          <td align="right" valign="top">
            
            
          </td>
        </tr>
      </table>
      
    </td>
  </tr>
<tr>
    <td width="15%" align="right" valign="top" class="summary">
      <span class="summary-type">node object</span>
    </td><td class="summary">
      <table width="100%" cellpadding="0" cellspacing="0" border="0">
        <tr>
          <td><span class="summary-sig"><a href="script-phyloFixedVar-module.html#readTree" class="summary-sig-name">readTree</a>(<span class="summary-sig-arg">file</span>)</span><br />
      Read a newick file and stock informations in a node object</td>
          <td align="right" valign="top">
            
            
          </td>
        </tr>
      </table>
      
    </td>
  </tr>
<tr>
    <td width="15%" align="right" valign="top" class="summary">
      <span class="summary-type">&nbsp;</span>
    </td><td class="summary">
      <table width="100%" cellpadding="0" cellspacing="0" border="0">
        <tr>
          <td><span class="summary-sig"><a href="script-phyloFixedVar-module.html#allNodesLeafs" class="summary-sig-name">allNodesLeafs</a>(<span class="summary-sig-arg">node</span>,
        <span class="summary-sig-arg">liste</span>)</span><br />
      For a given node, stock all its descendants name in a list (recursive
      function)</td>
          <td align="right" valign="top">
            
            
          </td>
        </tr>
      </table>
      
    </td>
  </tr>
<tr>
    <td width="15%" align="right" valign="top" class="summary">
      <span class="summary-type">&nbsp;</span>
    </td><td class="summary">
      <table width="100%" cellpadding="0" cellspacing="0" border="0">
        <tr>
          <td><span class="summary-sig"><a href="script-phyloFixedVar-module.html#groupCompare" class="summary-sig-name">groupCompare</a>(<span class="summary-sig-arg">node</span>,
        <span class="summary-sig-arg">dico</span>)</span><br />
      Complete a dictionnary which will contain a id as key and a list as 
      value.</td>
          <td align="right" valign="top">
            
            
          </td>
        </tr>
      </table>
      
    </td>
  </tr>
<tr>
    <td width="15%" align="right" valign="top" class="summary">
      <span class="summary-type">&nbsp;</span>
    </td><td class="summary">
      <table width="100%" cellpadding="0" cellspacing="0" border="0">
        <tr>
          <td><span class="summary-sig"><a href="script-phyloFixedVar-module.html#AllgroupCompare" class="summary-sig-name">AllgroupCompare</a>(<span class="summary-sig-arg">node</span>,
        <span class="summary-sig-arg">dico</span>)</span><br />
      Launch the groupCompare funtion on all sons of a given node</td>
          <td align="right" valign="top">
            
            
          </td>
        </tr>
      </table>
      
    </td>
  </tr>
<tr>
    <td width="15%" align="right" valign="top" class="summary">
      <span class="summary-type">&nbsp;</span>
    </td><td class="summary">
      <table width="100%" cellpadding="0" cellspacing="0" border="0">
        <tr>
          <td><span class="summary-sig"><a href="script-phyloFixedVar-module.html#addANN" class="summary-sig-name">addANN</a>(<span class="summary-sig-arg">objetNucl</span>,
        <span class="summary-sig-arg">annotation</span>)</span><br />
      Add annotation informations to a var object</td>
          <td align="right" valign="top">
            
            
          </td>
        </tr>
      </table>
      
    </td>
  </tr>
<tr>
    <td width="15%" align="right" valign="top" class="summary">
      <span class="summary-type">list</span>
    </td><td class="summary">
      <table width="100%" cellpadding="0" cellspacing="0" border="0">
        <tr>
          <td><span class="summary-sig"><a href="script-phyloFixedVar-module.html#readVCFfile" class="summary-sig-name">readVCFfile</a>(<span class="summary-sig-arg">file</span>)</span><br />
      Read a VCF file and make a list constituted by genome objects from 
      genome in the VCF</td>
          <td align="right" valign="top">
            
            
          </td>
        </tr>
      </table>
      
    </td>
  </tr>
<tr>
    <td width="15%" align="right" valign="top" class="summary">
      <span class="summary-type">dictionnary</span>
    </td><td class="summary">
      <table width="100%" cellpadding="0" cellspacing="0" border="0">
        <tr>
          <td><span class="summary-sig"><a href="script-phyloFixedVar-module.html#selectPosSensible" class="summary-sig-name">selectPosSensible</a>(<span class="summary-sig-arg">listeGenomes</span>)</span><br />
      Select all position which all genome from a list have the same 
      allele.</td>
          <td align="right" valign="top">
            
            
          </td>
        </tr>
      </table>
      
    </td>
  </tr>
<tr>
    <td width="15%" align="right" valign="top" class="summary">
      <span class="summary-type">list</span>
    </td><td class="summary">
      <table width="100%" cellpadding="0" cellspacing="0" border="0">
        <tr>
          <td><span class="summary-sig"><a href="script-phyloFixedVar-module.html#genomeName_to_genomeObjet" class="summary-sig-name">genomeName_to_genomeObjet</a>(<span class="summary-sig-arg">GenomesNames</span>,
        <span class="summary-sig-arg">GenomesObjet</span>)</span><br />
      From a list of genomes name, this fuction find the genome objects 
      associated to its name.</td>
          <td align="right" valign="top">
            
            
          </td>
        </tr>
      </table>
      
    </td>
  </tr>
<tr>
    <td width="15%" align="right" valign="top" class="summary">
      <span class="summary-type">dictionnary</span>
    </td><td class="summary">
      <table width="100%" cellpadding="0" cellspacing="0" border="0">
        <tr>
          <td><span class="summary-sig"><a href="script-phyloFixedVar-module.html#createDicoPos" class="summary-sig-name">createDicoPos</a>(<span class="summary-sig-arg">listeGenomesObjets</span>)</span><br />
      Make a dictionnary with all variants position and the all the 
      different alleles associated</td>
          <td align="right" valign="top">
            
            
          </td>
        </tr>
      </table>
      
    </td>
  </tr>
<tr>
    <td width="15%" align="right" valign="top" class="summary">
      <span class="summary-type">dictionnary</span>
    </td><td class="summary">
      <table width="100%" cellpadding="0" cellspacing="0" border="0">
        <tr>
          <td><span class="summary-sig"><a href="script-phyloFixedVar-module.html#selectPosSpecifique" class="summary-sig-name">selectPosSpecifique</a>(<span class="summary-sig-arg">dicoPosSensible</span>,
        <span class="summary-sig-arg">listeGenomesObjets</span>)</span><br />
      Select the alleles existing in the dictionnary and not present for 
      the genomes of the list (specificity filter)</td>
          <td align="right" valign="top">
            
            
          </td>
        </tr>
      </table>
      
    </td>
  </tr>
<tr>
    <td width="15%" align="right" valign="top" class="summary">
      <span class="summary-type">list</span>
    </td><td class="summary">
      <table width="100%" cellpadding="0" cellspacing="0" border="0">
        <tr>
          <td><span class="summary-sig"><a href="script-phyloFixedVar-module.html#allOtherGenomes" class="summary-sig-name">allOtherGenomes</a>(<span class="summary-sig-arg">Genomes_to_exclude</span>,
        <span class="summary-sig-arg">allGenomesObjets</span>)</span><br />
      Make a liste of genome name from a list of genome objects without the
      genomes name given in argument</td>
          <td align="right" valign="top">
            
            
          </td>
        </tr>
      </table>
      
    </td>
  </tr>
<tr>
    <td width="15%" align="right" valign="top" class="summary">
      <span class="summary-type">&nbsp;</span>
    </td><td class="summary">
      <table width="100%" cellpadding="0" cellspacing="0" border="0">
        <tr>
          <td><span class="summary-sig"><a href="script-phyloFixedVar-module.html#addNodeNameNewick" class="summary-sig-name">addNodeNameNewick</a>(<span class="summary-sig-arg">filename</span>,
        <span class="summary-sig-arg">outputFilename</span>)</span><br />
      Make a new newick fill with a label for each node.</td>
          <td align="right" valign="top">
            
            
          </td>
        </tr>
      </table>
      
    </td>
  </tr>
<tr>
    <td width="15%" align="right" valign="top" class="summary">
      <span class="summary-type">etree</span>
    </td><td class="summary">
      <table width="100%" cellpadding="0" cellspacing="0" border="0">
        <tr>
          <td><span class="summary-sig"><a href="script-phyloFixedVar-module.html#makeXML" class="summary-sig-name">makeXML</a>(<span class="summary-sig-arg">dico</span>,
        <span class="summary-sig-arg">GenomesObjet</span>)</span><br />
      Make a XML objet from a the comparaison groups list</td>
          <td align="right" valign="top">
            
            
          </td>
        </tr>
      </table>
      
    </td>
  </tr>
<tr>
    <td width="15%" align="right" valign="top" class="summary">
      <span class="summary-type">&nbsp;</span>
    </td><td class="summary">
      <table width="100%" cellpadding="0" cellspacing="0" border="0">
        <tr>
          <td><span class="summary-sig"><a name="main"></a><span class="summary-sig-name">main</span>()</span></td>
          <td align="right" valign="top">
            
            
          </td>
        </tr>
      </table>
      
    </td>
  </tr>
</table>
<!-- ==================== VARIABLES ==================== -->
<a name="section-Variables"></a>
<table class="summary" border="1" cellpadding="3"
       cellspacing="0" width="100%" bgcolor="white">
<tr bgcolor="#70b0f0" class="table-header">
  <td colspan="2" class="table-header">
    <table border="0" cellpadding="0" cellspacing="0" width="100%">
      <tr valign="top">
        <td align="left"><span class="table-header">Variables</span></td>
        <td align="right" valign="top"
         ><span class="options">[<a href="#section-Variables"
         class="privatelink" onclick="toggle_private();"
         >hide private</a>]</span></td>
      </tr>
    </table>
  </td>
</tr>
<tr>
    <td width="15%" align="right" valign="top" class="summary">
      <span class="summary-type">&nbsp;</span>
    </td><td class="summary">
        <a href="script-phyloFixedVar-module.html#__doc__" class="summary-name">__doc__</a> = <code title="&quot;&quot;&quot;
Extract all fixed variants of phylogeny nodes from a VCF annotated fil\
e. 

For each node from a phylogeny, 2 combination are make (one branch aga\
in the other and inversely). 
The script extract variants present in a branch and not existing in th\
e other branch (cousins). Homoplasy is possible at this step.
..."><code class="variable-ellipsis">...</code></code>
    </td>
  </tr>
</table>
<!-- ==================== FUNCTION DETAILS ==================== -->
<a name="section-FunctionDetails"></a>
<table class="details" border="1" cellpadding="3"
       cellspacing="0" width="100%" bgcolor="white">
<tr bgcolor="#70b0f0" class="table-header">
  <td colspan="2" class="table-header">
    <table border="0" cellpadding="0" cellspacing="0" width="100%">
      <tr valign="top">
        <td align="left"><span class="table-header">Function Details</span></td>
        <td align="right" valign="top"
         ><span class="options">[<a href="#section-FunctionDetails"
         class="privatelink" onclick="toggle_private();"
         >hide private</a>]</span></td>
      </tr>
    </table>
  </td>
</tr>
</table>
<a name="get_parser"></a>
<div>
<table class="details" border="1" cellpadding="3"
       cellspacing="0" width="100%" bgcolor="white">
<tr><td>
  <table width="100%" cellpadding="0" cellspacing="0" border="0">
  <tr valign="top"><td>
  <h3 class="epydoc"><span class="sig"><span class="sig-name">get_parser</span>()</span>
  </h3>
  </td><td align="right" valign="top"
    >&nbsp;
    </td>
  </tr></table>
  
  <p>Parse arguments</p>
  <dl class="fields">
    <dt>Returns: parser object</dt>
        <dd>arguments list</dd>
  </dl>
</td></tr></table>
</div>
<a name="printObjNucl"></a>
<div>
<table class="details" border="1" cellpadding="3"
       cellspacing="0" width="100%" bgcolor="white">
<tr><td>
  <table width="100%" cellpadding="0" cellspacing="0" border="0">
  <tr valign="top"><td>
  <h3 class="epydoc"><span class="sig"><span class="sig-name">printObjNucl</span>(<span class="sig-arg">objetNucl</span>)</span>
  </h3>
  </td><td align="right" valign="top"
    >&nbsp;
    </td>
  </tr></table>
  
  <p>Debug fonction, print all attributes of var object</p>
  <dl class="fields">
    <dt>Parameters:</dt>
    <dd><ul class="nomargin-top">
        <li><strong class="pname"><code>objetNucl</code></strong> (var object) - object var to print</li>
    </ul></dd>
  </dl>
</td></tr></table>
</div>
<a name="createNode"></a>
<div>
<table class="details" border="1" cellpadding="3"
       cellspacing="0" width="100%" bgcolor="white">
<tr><td>
  <table width="100%" cellpadding="0" cellspacing="0" border="0">
  <tr valign="top"><td>
  <h3 class="epydoc"><span class="sig"><span class="sig-name">createNode</span>()</span>
  </h3>
  </td><td align="right" valign="top"
    >&nbsp;
    </td>
  </tr></table>
  
  <p>Create a new empty node object</p>
  <dl class="fields">
    <dt>Returns: node object</dt>
        <dd>empty node object</dd>
  </dl>
</td></tr></table>
</div>
<a name="readTree"></a>
<div>
<table class="details" border="1" cellpadding="3"
       cellspacing="0" width="100%" bgcolor="white">
<tr><td>
  <table width="100%" cellpadding="0" cellspacing="0" border="0">
  <tr valign="top"><td>
  <h3 class="epydoc"><span class="sig"><span class="sig-name">readTree</span>(<span class="sig-arg">file</span>)</span>
  </h3>
  </td><td align="right" valign="top"
    >&nbsp;
    </td>
  </tr></table>
  
  <p>Read a newick file and stock informations in a node object</p>
  <dl class="fields">
    <dt>Parameters:</dt>
    <dd><ul class="nomargin-top">
        <li><strong class="pname"><code>file</code></strong> (string) - newick file name</li>
    </ul></dd>
    <dt>Returns: node object</dt>
        <dd>root of the phylogeny tree</dd>
  </dl>
</td></tr></table>
</div>
<a name="allNodesLeafs"></a>
<div>
<table class="details" border="1" cellpadding="3"
       cellspacing="0" width="100%" bgcolor="white">
<tr><td>
  <table width="100%" cellpadding="0" cellspacing="0" border="0">
  <tr valign="top"><td>
  <h3 class="epydoc"><span class="sig"><span class="sig-name">allNodesLeafs</span>(<span class="sig-arg">node</span>,
        <span class="sig-arg">liste</span>)</span>
  </h3>
  </td><td align="right" valign="top"
    >&nbsp;
    </td>
  </tr></table>
  
  <p>For a given node, stock all its descendants name in a list (recursive 
  function)</p>
  <dl class="fields">
    <dt>Parameters:</dt>
    <dd><ul class="nomargin-top">
        <li><strong class="pname"><code>node</code></strong> (node object) - node object which descendants names will be found</li>
        <li><strong class="pname"><code>liste</code></strong> (list) - empty liste that will contain genomes name</li>
    </ul></dd>
  </dl>
</td></tr></table>
</div>
<a name="groupCompare"></a>
<div>
<table class="details" border="1" cellpadding="3"
       cellspacing="0" width="100%" bgcolor="white">
<tr><td>
  <table width="100%" cellpadding="0" cellspacing="0" border="0">
  <tr valign="top"><td>
  <h3 class="epydoc"><span class="sig"><span class="sig-name">groupCompare</span>(<span class="sig-arg">node</span>,
        <span class="sig-arg">dico</span>)</span>
  </h3>
  </td><td align="right" valign="top"
    >&nbsp;
    </td>
  </tr></table>
  
  <p>Complete a dictionnary which will contain a id as key and a list as 
  value.      The list is composed by 2 element :</p>
  <ul>
    <li>
      the first element is a list of all descendants genomes name of a node
      object,
    </li>
    <li>
      the second element is a list of all cousin(s) genomes names.  \
    </li>
  </ul>
  <p>This function is applied on all sons of the given node.</p>
  <dl class="fields">
    <dt>Parameters:</dt>
    <dd><ul class="nomargin-top">
        <li><strong class="pname"><code>node</code></strong> (node object) - kin node object to process</li>
        <li><strong class="pname"><code>dico</code></strong> (dictionnary) - dictionnary to complete</li>
    </ul></dd>
  </dl>
</td></tr></table>
</div>
<a name="AllgroupCompare"></a>
<div>
<table class="details" border="1" cellpadding="3"
       cellspacing="0" width="100%" bgcolor="white">
<tr><td>
  <table width="100%" cellpadding="0" cellspacing="0" border="0">
  <tr valign="top"><td>
  <h3 class="epydoc"><span class="sig"><span class="sig-name">AllgroupCompare</span>(<span class="sig-arg">node</span>,
        <span class="sig-arg">dico</span>)</span>
  </h3>
  </td><td align="right" valign="top"
    >&nbsp;
    </td>
  </tr></table>
  
  <p>Launch the groupCompare funtion on all sons of a given node</p>
  <dl class="fields">
    <dt>Parameters:</dt>
    <dd><ul class="nomargin-top">
        <li><strong class="pname"><code>node</code></strong> (node object) - kin node object to process</li>
        <li><strong class="pname"><code>dico</code></strong> (dictionnary) - dictionnary to complete</li>
    </ul></dd>
  </dl>
</td></tr></table>
</div>
<a name="addANN"></a>
<div>
<table class="details" border="1" cellpadding="3"
       cellspacing="0" width="100%" bgcolor="white">
<tr><td>
  <table width="100%" cellpadding="0" cellspacing="0" border="0">
  <tr valign="top"><td>
  <h3 class="epydoc"><span class="sig"><span class="sig-name">addANN</span>(<span class="sig-arg">objetNucl</span>,
        <span class="sig-arg">annotation</span>)</span>
  </h3>
  </td><td align="right" valign="top"
    >&nbsp;
    </td>
  </tr></table>
  
  <p>Add annotation informations to a var object</p>
  <dl class="fields">
    <dt>Parameters:</dt>
    <dd><ul class="nomargin-top">
        <li><strong class="pname"><code>objetNucl</code></strong> (var object) - var object to complete</li>
        <li><strong class="pname"><code>annotation</code></strong> (string) - for one line in a VCF file, the info column</li>
    </ul></dd>
  </dl>
</td></tr></table>
</div>
<a name="readVCFfile"></a>
<div>
<table class="details" border="1" cellpadding="3"
       cellspacing="0" width="100%" bgcolor="white">
<tr><td>
  <table width="100%" cellpadding="0" cellspacing="0" border="0">
  <tr valign="top"><td>
  <h3 class="epydoc"><span class="sig"><span class="sig-name">readVCFfile</span>(<span class="sig-arg">file</span>)</span>
  </h3>
  </td><td align="right" valign="top"
    >&nbsp;
    </td>
  </tr></table>
  
  <p>Read a VCF file and make a list constituted by genome objects from 
  genome in the VCF</p>
  <dl class="fields">
    <dt>Parameters:</dt>
    <dd><ul class="nomargin-top">
        <li><strong class="pname"><code>file</code></strong> (file) - VCF file</li>
    </ul></dd>
    <dt>Returns: list</dt>
        <dd>list of genome objects</dd>
  </dl>
</td></tr></table>
</div>
<a name="selectPosSensible"></a>
<div>
<table class="details" border="1" cellpadding="3"
       cellspacing="0" width="100%" bgcolor="white">
<tr><td>
  <table width="100%" cellpadding="0" cellspacing="0" border="0">
  <tr valign="top"><td>
  <h3 class="epydoc"><span class="sig"><span class="sig-name">selectPosSensible</span>(<span class="sig-arg">listeGenomes</span>)</span>
  </h3>
  </td><td align="right" valign="top"
    >&nbsp;
    </td>
  </tr></table>
  
  <p>Select all position which all genome from a list have the same allele.
  Not necessary for only one genome.</p>
  <dl class="fields">
    <dt>Parameters:</dt>
    <dd><ul class="nomargin-top">
        <li><strong class="pname"><code>listeGenomes</code></strong> (list) - list of genome objects</li>
    </ul></dd>
    <dt>Returns: dictionnary</dt>
        <dd>dictionnary with the position as key and the nucleic allele for 
          value</dd>
  </dl>
</td></tr></table>
</div>
<a name="genomeName_to_genomeObjet"></a>
<div>
<table class="details" border="1" cellpadding="3"
       cellspacing="0" width="100%" bgcolor="white">
<tr><td>
  <table width="100%" cellpadding="0" cellspacing="0" border="0">
  <tr valign="top"><td>
  <h3 class="epydoc"><span class="sig"><span class="sig-name">genomeName_to_genomeObjet</span>(<span class="sig-arg">GenomesNames</span>,
        <span class="sig-arg">GenomesObjet</span>)</span>
  </h3>
  </td><td align="right" valign="top"
    >&nbsp;
    </td>
  </tr></table>
  
  <p>From a list of genomes name, this fuction find the genome objects 
  associated to its name.</p>
  <dl class="fields">
    <dt>Parameters:</dt>
    <dd><ul class="nomargin-top">
        <li><strong class="pname"><code>GenomesNames</code></strong> (list) - list of genome names</li>
        <li><strong class="pname"><code>GenomesObjet</code></strong> (list) - list of genome objects</li>
    </ul></dd>
    <dt>Returns: list</dt>
        <dd>list of genome objects associated to the genome names of the 
          GenomesNames list</dd>
  </dl>
</td></tr></table>
</div>
<a name="createDicoPos"></a>
<div>
<table class="details" border="1" cellpadding="3"
       cellspacing="0" width="100%" bgcolor="white">
<tr><td>
  <table width="100%" cellpadding="0" cellspacing="0" border="0">
  <tr valign="top"><td>
  <h3 class="epydoc"><span class="sig"><span class="sig-name">createDicoPos</span>(<span class="sig-arg">listeGenomesObjets</span>)</span>
  </h3>
  </td><td align="right" valign="top"
    >&nbsp;
    </td>
  </tr></table>
  
  <p>Make a dictionnary with all variants position and the all the 
  different alleles associated</p>
  <dl class="fields">
    <dt>Parameters:</dt>
    <dd><ul class="nomargin-top">
        <li><strong class="pname"><code>listeGenomesObjets</code></strong> (list) - list of genome objects</li>
    </ul></dd>
    <dt>Returns: dictionnary</dt>
        <dd>dictionnary with position as key and the list of the different 
          alleles at this position for value</dd>
  </dl>
</td></tr></table>
</div>
<a name="selectPosSpecifique"></a>
<div>
<table class="details" border="1" cellpadding="3"
       cellspacing="0" width="100%" bgcolor="white">
<tr><td>
  <table width="100%" cellpadding="0" cellspacing="0" border="0">
  <tr valign="top"><td>
  <h3 class="epydoc"><span class="sig"><span class="sig-name">selectPosSpecifique</span>(<span class="sig-arg">dicoPosSensible</span>,
        <span class="sig-arg">listeGenomesObjets</span>)</span>
  </h3>
  </td><td align="right" valign="top"
    >&nbsp;
    </td>
  </tr></table>
  
  <p>Select the alleles existing in the dictionnary and not present for the
  genomes of the list (specificity filter)</p>
  <dl class="fields">
    <dt>Parameters:</dt>
    <dd><ul class="nomargin-top">
        <li><strong class="pname"><code>dicoPosSensible</code></strong> (dictionnary) - dictionnary with positions as key and the allele corresponding as
          value</li>
        <li><strong class="pname"><code>listeGenomesObjets</code></strong> (list) - list of genome objects</li>
    </ul></dd>
    <dt>Returns: dictionnary</dt>
        <dd>dictionnary with position specifique as key and the allele 
          corresponding as value</dd>
  </dl>
</td></tr></table>
</div>
<a name="allOtherGenomes"></a>
<div>
<table class="details" border="1" cellpadding="3"
       cellspacing="0" width="100%" bgcolor="white">
<tr><td>
  <table width="100%" cellpadding="0" cellspacing="0" border="0">
  <tr valign="top"><td>
  <h3 class="epydoc"><span class="sig"><span class="sig-name">allOtherGenomes</span>(<span class="sig-arg">Genomes_to_exclude</span>,
        <span class="sig-arg">allGenomesObjets</span>)</span>
  </h3>
  </td><td align="right" valign="top"
    >&nbsp;
    </td>
  </tr></table>
  
  <p>Make a liste of genome name from a list of genome objects without the 
  genomes name given in argument</p>
  <dl class="fields">
    <dt>Parameters:</dt>
    <dd><ul class="nomargin-top">
        <li><strong class="pname"><code>Genomes_to_exclude</code></strong> (list) - list of genomes name to exclude</li>
        <li><strong class="pname"><code>allGenomesObjets</code></strong> (list) - list of genome objects</li>
    </ul></dd>
    <dt>Returns: list</dt>
        <dd>a list of genomes name without the given name</dd>
  </dl>
</td></tr></table>
</div>
<a name="addNodeNameNewick"></a>
<div>
<table class="details" border="1" cellpadding="3"
       cellspacing="0" width="100%" bgcolor="white">
<tr><td>
  <table width="100%" cellpadding="0" cellspacing="0" border="0">
  <tr valign="top"><td>
  <h3 class="epydoc"><span class="sig"><span class="sig-name">addNodeNameNewick</span>(<span class="sig-arg">filename</span>,
        <span class="sig-arg">outputFilename</span>)</span>
  </h3>
  </td><td align="right" valign="top"
    >&nbsp;
    </td>
  </tr></table>
  
  <p>Make a new newick fill with a label for each node. The label is 
  created like this :</p>
  <ul>
    <li>
      N + the number of the node
    </li>
    <li>
      [CX-CY] with X and Y the number of combination
    </li>
  </ul>
  <dl class="fields">
    <dt>Parameters:</dt>
    <dd><ul class="nomargin-top">
        <li><strong class="pname"><code>filename</code></strong> (string) - newick input file name</li>
        <li><strong class="pname"><code>outputFilename</code></strong> (string) - newick output file name</li>
    </ul></dd>
  </dl>
</td></tr></table>
</div>
<a name="makeXML"></a>
<div>
<table class="details" border="1" cellpadding="3"
       cellspacing="0" width="100%" bgcolor="white">
<tr><td>
  <table width="100%" cellpadding="0" cellspacing="0" border="0">
  <tr valign="top"><td>
  <h3 class="epydoc"><span class="sig"><span class="sig-name">makeXML</span>(<span class="sig-arg">dico</span>,
        <span class="sig-arg">GenomesObjet</span>)</span>
  </h3>
  </td><td align="right" valign="top"
    >&nbsp;
    </td>
  </tr></table>
  
  <p>Make a XML objet from a the comparaison groups list</p>
  <dl class="fields">
    <dt>Parameters:</dt>
    <dd><ul class="nomargin-top">
        <li><strong class="pname"><code>dico</code></strong> (string) - dictionnary obtained with the AllgroupCompare function</li>
        <li><strong class="pname"><code>GenomesObjet</code></strong> (list) - list of genome objects</li>
    </ul></dd>
    <dt>Returns: etree</dt>
        <dd>root of the XML tree</dd>
  </dl>
</td></tr></table>
</div>
<br />
<!-- ==================== VARIABLES DETAILS ==================== -->
<a name="section-VariablesDetails"></a>
<table class="details" border="1" cellpadding="3"
       cellspacing="0" width="100%" bgcolor="white">
<tr bgcolor="#70b0f0" class="table-header">
  <td colspan="2" class="table-header">
    <table border="0" cellpadding="0" cellspacing="0" width="100%">
      <tr valign="top">
        <td align="left"><span class="table-header">Variables Details</span></td>
        <td align="right" valign="top"
         ><span class="options">[<a href="#section-VariablesDetails"
         class="privatelink" onclick="toggle_private();"
         >hide private</a>]</span></td>
      </tr>
    </table>
  </td>
</tr>
</table>
<a name="__doc__"></a>
<div>
<table class="details" border="1" cellpadding="3"
       cellspacing="0" width="100%" bgcolor="white">
<tr><td>
  <h3 class="epydoc">__doc__</h3>
  
  <dl class="fields">
  </dl>
  <dl class="fields">
    <dt>Value:</dt>
      <dd><table><tr><td><pre class="variable">
&quot;&quot;&quot;
Extract all fixed variants of phylogeny nodes from a VCF annotated fil<span class="variable-linewrap"><img src="crarr.png" alt="\" /></span>
e. 

For each node from a phylogeny, 2 combination are make (one branch aga<span class="variable-linewrap"><img src="crarr.png" alt="\" /></span>
in the other and inversely). 
The script extract variants present in a branch and not existing in th<span class="variable-linewrap"><img src="crarr.png" alt="\" /></span>
e other branch (cousins). Homoplasy is possible at this step.
<code class="variable-ellipsis">...</code>
</pre></td></tr></table>
</dd>
  </dl>
</td></tr></table>
</div>
<br />
<!-- ==================== NAVIGATION BAR ==================== -->
<table class="navbar" border="0" width="100%" cellpadding="0"
       bgcolor="#a0c0ff" cellspacing="0">
  <tr valign="middle">

  <!-- Tree link -->
      <th>&nbsp;&nbsp;&nbsp;<a
        href="module-tree.html">Trees</a>&nbsp;&nbsp;&nbsp;</th>

  <!-- Index link -->
      <th>&nbsp;&nbsp;&nbsp;<a
        href="identifier-index.html">Indices</a>&nbsp;&nbsp;&nbsp;</th>

  <!-- Help link -->
      <th>&nbsp;&nbsp;&nbsp;<a
        href="help.html">Help</a>&nbsp;&nbsp;&nbsp;</th>

      <th class="navbar" width="100%"></th>
  </tr>
</table>
<table border="0" cellpadding="0" cellspacing="0" width="100%%">
  <tr>
    <td align="left" class="footer">
    Generated by Epydoc 3.0.1 on Wed Sep 14 15:13:53 2016
    </td>
    <td align="right" class="footer">
      <a target="mainFrame" href="http://epydoc.sourceforge.net"
        >http://epydoc.sourceforge.net</a>
    </td>
  </tr>
</table>

<script type="text/javascript">
  <!--
  // Private objects are initially displayed (because if
  // javascript is turned off then we want them to be
  // visible); but by default, we want to hide them.  So hide
  // them unless we have a cookie that says to show them.
  checkCookie();
  // -->
</script>
</body>
</html>
