%
% API Documentation for API Documentation
% Script script-phyloFixedVar
%
% Generated by epydoc 3.0.1
% [Wed Sep 14 15:14:04 2016]
%

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
%%                          Module Description                           %%
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

    \index{script-phyloFixedVar \textit{(script)}|(}
\section{Script script-phyloFixedVar}

    \label{script-phyloFixedVar}

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
%%                               Functions                               %%
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

  \subsection{Functions}

    \label{script-phyloFixedVar:get_parser}
    \index{script-phyloFixedVar \textit{(script)}!script-phyloFixedVar.get\_parser \textit{(function)}}

    \vspace{0.5ex}

\hspace{.8\funcindent}\begin{boxedminipage}{\funcwidth}

    \raggedright \textbf{get\_parser}()

    \vspace{-1.5ex}

    \rule{\textwidth}{0.5\fboxrule}
\setlength{\parskip}{2ex}
    Parse arguments

\setlength{\parskip}{1ex}
      \textbf{Return Value}
    \vspace{-1ex}

      \begin{quote}
      arguments list

      {\it (type=parser object)}

      \end{quote}

    \end{boxedminipage}

    \label{script-phyloFixedVar:printObjNucl}
    \index{script-phyloFixedVar \textit{(script)}!script-phyloFixedVar.printObjNucl \textit{(function)}}

    \vspace{0.5ex}

\hspace{.8\funcindent}\begin{boxedminipage}{\funcwidth}

    \raggedright \textbf{printObjNucl}(\textit{objetNucl})

    \vspace{-1.5ex}

    \rule{\textwidth}{0.5\fboxrule}
\setlength{\parskip}{2ex}
    Debug fonction, print all attributes of var object

\setlength{\parskip}{1ex}
      \textbf{Parameters}
      \vspace{-1ex}

      \begin{quote}
        \begin{Ventry}{xxxxxxxxx}

          \item[objetNucl]

          object var to print

            {\it (type=var object)}

        \end{Ventry}

      \end{quote}

    \end{boxedminipage}

    \label{script-phyloFixedVar:createNode}
    \index{script-phyloFixedVar \textit{(script)}!script-phyloFixedVar.createNode \textit{(function)}}

    \vspace{0.5ex}

\hspace{.8\funcindent}\begin{boxedminipage}{\funcwidth}

    \raggedright \textbf{createNode}()

    \vspace{-1.5ex}

    \rule{\textwidth}{0.5\fboxrule}
\setlength{\parskip}{2ex}
    Create a new empty node object

\setlength{\parskip}{1ex}
      \textbf{Return Value}
    \vspace{-1ex}

      \begin{quote}
      empty node object

      {\it (type=node object)}

      \end{quote}

    \end{boxedminipage}

    \label{script-phyloFixedVar:readTree}
    \index{script-phyloFixedVar \textit{(script)}!script-phyloFixedVar.readTree \textit{(function)}}

    \vspace{0.5ex}

\hspace{.8\funcindent}\begin{boxedminipage}{\funcwidth}

    \raggedright \textbf{readTree}(\textit{file})

    \vspace{-1.5ex}

    \rule{\textwidth}{0.5\fboxrule}
\setlength{\parskip}{2ex}
    Read a newick file and stock informations in a node object

\setlength{\parskip}{1ex}
      \textbf{Parameters}
      \vspace{-1ex}

      \begin{quote}
        \begin{Ventry}{xxxx}

          \item[file]

          newick file name

            {\it (type=string)}

        \end{Ventry}

      \end{quote}

      \textbf{Return Value}
    \vspace{-1ex}

      \begin{quote}
      root of the phylogeny tree

      {\it (type=node object)}

      \end{quote}

    \end{boxedminipage}

    \label{script-phyloFixedVar:allNodesLeafs}
    \index{script-phyloFixedVar \textit{(script)}!script-phyloFixedVar.allNodesLeafs \textit{(function)}}

    \vspace{0.5ex}

\hspace{.8\funcindent}\begin{boxedminipage}{\funcwidth}

    \raggedright \textbf{allNodesLeafs}(\textit{node}, \textit{liste})

    \vspace{-1.5ex}

    \rule{\textwidth}{0.5\fboxrule}
\setlength{\parskip}{2ex}
    For a given node, stock all its descendants name in a list (recursive 
    function)

\setlength{\parskip}{1ex}
      \textbf{Parameters}
      \vspace{-1ex}

      \begin{quote}
        \begin{Ventry}{xxxxx}

          \item[node]

          node object which descendants names will be found

            {\it (type=node object)}

          \item[liste]

          empty liste that will contain genomes name

            {\it (type=list)}

        \end{Ventry}

      \end{quote}

    \end{boxedminipage}

    \label{script-phyloFixedVar:groupCompare}
    \index{script-phyloFixedVar \textit{(script)}!script-phyloFixedVar.groupCompare \textit{(function)}}

    \vspace{0.5ex}

\hspace{.8\funcindent}\begin{boxedminipage}{\funcwidth}

    \raggedright \textbf{groupCompare}(\textit{node}, \textit{dico})

    \vspace{-1.5ex}

    \rule{\textwidth}{0.5\fboxrule}
\setlength{\parskip}{2ex}
    Complete a dictionnary which will contain a id as key and a list as 
    value.      The list is composed by 2 element :

    \begin{itemize}
    \setlength{\parskip}{0.6ex}
      \item the first element is a list of all descendants genomes name of a 
        node object,

      \item the second element is a list of all cousin(s) genomes names.  
        {\textbackslash}

    \end{itemize}

    This function is applied on all sons of the given node.

\setlength{\parskip}{1ex}
      \textbf{Parameters}
      \vspace{-1ex}

      \begin{quote}
        \begin{Ventry}{xxxx}

          \item[node]

          kin node object to process

            {\it (type=node object)}

          \item[dico]

          dictionnary to complete

            {\it (type=dictionnary)}

        \end{Ventry}

      \end{quote}

    \end{boxedminipage}

    \label{script-phyloFixedVar:AllgroupCompare}
    \index{script-phyloFixedVar \textit{(script)}!script-phyloFixedVar.AllgroupCompare \textit{(function)}}

    \vspace{0.5ex}

\hspace{.8\funcindent}\begin{boxedminipage}{\funcwidth}

    \raggedright \textbf{AllgroupCompare}(\textit{node}, \textit{dico})

    \vspace{-1.5ex}

    \rule{\textwidth}{0.5\fboxrule}
\setlength{\parskip}{2ex}
    Launch the groupCompare funtion on all sons of a given node

\setlength{\parskip}{1ex}
      \textbf{Parameters}
      \vspace{-1ex}

      \begin{quote}
        \begin{Ventry}{xxxx}

          \item[node]

          kin node object to process

            {\it (type=node object)}

          \item[dico]

          dictionnary to complete

            {\it (type=dictionnary)}

        \end{Ventry}

      \end{quote}

    \end{boxedminipage}

    \label{script-phyloFixedVar:addANN}
    \index{script-phyloFixedVar \textit{(script)}!script-phyloFixedVar.addANN \textit{(function)}}

    \vspace{0.5ex}

\hspace{.8\funcindent}\begin{boxedminipage}{\funcwidth}

    \raggedright \textbf{addANN}(\textit{objetNucl}, \textit{annotation})

    \vspace{-1.5ex}

    \rule{\textwidth}{0.5\fboxrule}
\setlength{\parskip}{2ex}
    Add annotation informations to a var object

\setlength{\parskip}{1ex}
      \textbf{Parameters}
      \vspace{-1ex}

      \begin{quote}
        \begin{Ventry}{xxxxxxxxxx}

          \item[objetNucl]

          var object to complete

            {\it (type=var object)}

          \item[annotation]

          for one line in a VCF file, the info column

            {\it (type=string)}

        \end{Ventry}

      \end{quote}

    \end{boxedminipage}

    \label{script-phyloFixedVar:readVCFfile}
    \index{script-phyloFixedVar \textit{(script)}!script-phyloFixedVar.readVCFfile \textit{(function)}}

    \vspace{0.5ex}

\hspace{.8\funcindent}\begin{boxedminipage}{\funcwidth}

    \raggedright \textbf{readVCFfile}(\textit{file})

    \vspace{-1.5ex}

    \rule{\textwidth}{0.5\fboxrule}
\setlength{\parskip}{2ex}
    Read a VCF file and make a list constituted by genome objects from 
    genome in the VCF

\setlength{\parskip}{1ex}
      \textbf{Parameters}
      \vspace{-1ex}

      \begin{quote}
        \begin{Ventry}{xxxx}

          \item[file]

          VCF file

            {\it (type=file)}

        \end{Ventry}

      \end{quote}

      \textbf{Return Value}
    \vspace{-1ex}

      \begin{quote}
      list of genome objects

      {\it (type=list)}

      \end{quote}

    \end{boxedminipage}

    \label{script-phyloFixedVar:selectPosSensible}
    \index{script-phyloFixedVar \textit{(script)}!script-phyloFixedVar.selectPosSensible \textit{(function)}}

    \vspace{0.5ex}

\hspace{.8\funcindent}\begin{boxedminipage}{\funcwidth}

    \raggedright \textbf{selectPosSensible}(\textit{listeGenomes})

    \vspace{-1.5ex}

    \rule{\textwidth}{0.5\fboxrule}
\setlength{\parskip}{2ex}
    Select all position which all genome from a list have the same allele. 
    Not necessary for only one genome.

\setlength{\parskip}{1ex}
      \textbf{Parameters}
      \vspace{-1ex}

      \begin{quote}
        \begin{Ventry}{xxxxxxxxxxxx}

          \item[listeGenomes]

          list of genome objects

            {\it (type=list)}

        \end{Ventry}

      \end{quote}

      \textbf{Return Value}
    \vspace{-1ex}

      \begin{quote}
      dictionnary with the position as key and the nucleic allele for value

      {\it (type=dictionnary)}

      \end{quote}

    \end{boxedminipage}

    \label{script-phyloFixedVar:genomeName_to_genomeObjet}
    \index{script-phyloFixedVar \textit{(script)}!script-phyloFixedVar.genomeName\_to\_genomeObjet \textit{(function)}}

    \vspace{0.5ex}

\hspace{.8\funcindent}\begin{boxedminipage}{\funcwidth}

    \raggedright \textbf{genomeName\_to\_genomeObjet}(\textit{GenomesNames}, \textit{GenomesObjet})

    \vspace{-1.5ex}

    \rule{\textwidth}{0.5\fboxrule}
\setlength{\parskip}{2ex}
    From a list of genomes name, this fuction find the genome objects 
    associated to its name.

\setlength{\parskip}{1ex}
      \textbf{Parameters}
      \vspace{-1ex}

      \begin{quote}
        \begin{Ventry}{xxxxxxxxxxxx}

          \item[GenomesNames]

          list of genome names

            {\it (type=list)}

          \item[GenomesObjet]

          list of genome objects

            {\it (type=list)}

        \end{Ventry}

      \end{quote}

      \textbf{Return Value}
    \vspace{-1ex}

      \begin{quote}
      list of genome objects associated to the genome names of the 
      GenomesNames list

      {\it (type=list)}

      \end{quote}

    \end{boxedminipage}

    \label{script-phyloFixedVar:createDicoPos}
    \index{script-phyloFixedVar \textit{(script)}!script-phyloFixedVar.createDicoPos \textit{(function)}}

    \vspace{0.5ex}

\hspace{.8\funcindent}\begin{boxedminipage}{\funcwidth}

    \raggedright \textbf{createDicoPos}(\textit{listeGenomesObjets})

    \vspace{-1.5ex}

    \rule{\textwidth}{0.5\fboxrule}
\setlength{\parskip}{2ex}
    Make a dictionnary with all variants position and the all the different
    alleles associated

\setlength{\parskip}{1ex}
      \textbf{Parameters}
      \vspace{-1ex}

      \begin{quote}
        \begin{Ventry}{xxxxxxxxxxxxxxxxxx}

          \item[listeGenomesObjets]

          list of genome objects

            {\it (type=list)}

        \end{Ventry}

      \end{quote}

      \textbf{Return Value}
    \vspace{-1ex}

      \begin{quote}
      dictionnary with position as key and the list of the different 
      alleles at this position for value

      {\it (type=dictionnary)}

      \end{quote}

    \end{boxedminipage}

    \label{script-phyloFixedVar:selectPosSpecifique}
    \index{script-phyloFixedVar \textit{(script)}!script-phyloFixedVar.selectPosSpecifique \textit{(function)}}

    \vspace{0.5ex}

\hspace{.8\funcindent}\begin{boxedminipage}{\funcwidth}

    \raggedright \textbf{selectPosSpecifique}(\textit{dicoPosSensible}, \textit{listeGenomesObjets})

    \vspace{-1.5ex}

    \rule{\textwidth}{0.5\fboxrule}
\setlength{\parskip}{2ex}
    Select the alleles existing in the dictionnary and not present for the 
    genomes of the list (specificity filter)

\setlength{\parskip}{1ex}
      \textbf{Parameters}
      \vspace{-1ex}

      \begin{quote}
        \begin{Ventry}{xxxxxxxxxxxxxxxxxx}

          \item[dicoPosSensible]

          dictionnary with positions as key and the allele corresponding as
          value

            {\it (type=dictionnary)}

          \item[listeGenomesObjets]

          list of genome objects

            {\it (type=list)}

        \end{Ventry}

      \end{quote}

      \textbf{Return Value}
    \vspace{-1ex}

      \begin{quote}
      dictionnary with position specifique as key and the allele 
      corresponding as value

      {\it (type=dictionnary)}

      \end{quote}

    \end{boxedminipage}

    \label{script-phyloFixedVar:allOtherGenomes}
    \index{script-phyloFixedVar \textit{(script)}!script-phyloFixedVar.allOtherGenomes \textit{(function)}}

    \vspace{0.5ex}

\hspace{.8\funcindent}\begin{boxedminipage}{\funcwidth}

    \raggedright \textbf{allOtherGenomes}(\textit{Genomes\_to\_exclude}, \textit{allGenomesObjets})

    \vspace{-1.5ex}

    \rule{\textwidth}{0.5\fboxrule}
\setlength{\parskip}{2ex}
    Make a liste of genome name from a list of genome objects without the 
    genomes name given in argument

\setlength{\parskip}{1ex}
      \textbf{Parameters}
      \vspace{-1ex}

      \begin{quote}
        \begin{Ventry}{xxxxxxxxxxxxxxxxxx}

          \item[Genomes\_to\_exclude]

          list of genomes name to exclude

            {\it (type=list)}

          \item[allGenomesObjets]

          list of genome objects

            {\it (type=list)}

        \end{Ventry}

      \end{quote}

      \textbf{Return Value}
    \vspace{-1ex}

      \begin{quote}
      a list of genomes name without the given name

      {\it (type=list)}

      \end{quote}

    \end{boxedminipage}

    \label{script-phyloFixedVar:addNodeNameNewick}
    \index{script-phyloFixedVar \textit{(script)}!script-phyloFixedVar.addNodeNameNewick \textit{(function)}}

    \vspace{0.5ex}

\hspace{.8\funcindent}\begin{boxedminipage}{\funcwidth}

    \raggedright \textbf{addNodeNameNewick}(\textit{filename}, \textit{outputFilename})

    \vspace{-1.5ex}

    \rule{\textwidth}{0.5\fboxrule}
\setlength{\parskip}{2ex}
    Make a new newick fill with a label for each node. The label is created
    like this :

    \begin{itemize}
    \setlength{\parskip}{0.6ex}
      \item N + the number of the node

      \item [CX-CY] with X and Y the number of combination

    \end{itemize}

\setlength{\parskip}{1ex}
      \textbf{Parameters}
      \vspace{-1ex}

      \begin{quote}
        \begin{Ventry}{xxxxxxxxxxxxxx}

          \item[filename]

          newick input file name

            {\it (type=string)}

          \item[outputFilename]

          newick output file name

            {\it (type=string)}

        \end{Ventry}

      \end{quote}

    \end{boxedminipage}

    \label{script-phyloFixedVar:makeXML}
    \index{script-phyloFixedVar \textit{(script)}!script-phyloFixedVar.makeXML \textit{(function)}}

    \vspace{0.5ex}

\hspace{.8\funcindent}\begin{boxedminipage}{\funcwidth}

    \raggedright \textbf{makeXML}(\textit{dico}, \textit{GenomesObjet})

    \vspace{-1.5ex}

    \rule{\textwidth}{0.5\fboxrule}
\setlength{\parskip}{2ex}
    Make a XML objet from a the comparaison groups list

\setlength{\parskip}{1ex}
      \textbf{Parameters}
      \vspace{-1ex}

      \begin{quote}
        \begin{Ventry}{xxxxxxxxxxxx}

          \item[dico]

          dictionnary obtained with the AllgroupCompare function

            {\it (type=string)}

          \item[GenomesObjet]

          list of genome objects

            {\it (type=list)}

        \end{Ventry}

      \end{quote}

      \textbf{Return Value}
    \vspace{-1ex}

      \begin{quote}
      root of the XML tree

      {\it (type=etree)}

      \end{quote}

    \end{boxedminipage}

    \label{script-phyloFixedVar:main}
    \index{script-phyloFixedVar \textit{(script)}!script-phyloFixedVar.main \textit{(function)}}

    \vspace{0.5ex}

\hspace{.8\funcindent}\begin{boxedminipage}{\funcwidth}

    \raggedright \textbf{main}()

\setlength{\parskip}{2ex}
\setlength{\parskip}{1ex}
    \end{boxedminipage}


%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
%%                               Variables                               %%
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

  \subsection{Variables}

    \vspace{-1cm}
\hspace{\varindent}\begin{longtable}{|p{\varnamewidth}|p{\vardescrwidth}|l}
\cline{1-2}
\cline{1-2} \centering \textbf{Name} & \centering \textbf{Description}& \\
\cline{1-2}
\endhead\cline{1-2}\multicolumn{3}{r}{\small\textit{continued on next page}}\\\endfoot\cline{1-2}
\endlastfoot\raggedright \_\-\_\-d\-o\-c\-\_\-\_\- & \raggedright \textbf{Value:} 
{\tt \texttt{...}}&\\
\cline{1-2}
\end{longtable}


%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
%%                           Class Description                           %%
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

    \index{script-phyloFixedVar \textit{(script)}!script-phyloFixedVar.var \textit{(class)}|(}
\subsection{Class var}

    \label{script-phyloFixedVar:var}
\begin{tabular}{cccccc}
% Line for object, linespec=[False]
\multicolumn{2}{r}{\settowidth{\BCL}{object}\multirow{2}{\BCL}{object}}
&&
  \\\cline{3-3}
  &&\multicolumn{1}{c|}{}
&&
  \\
&&\multicolumn{2}{l}{\textbf{script-phyloFixedVar.var}}
\end{tabular}

Create var object to stock variant information


%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
%%                                Methods                                %%
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

  \subsubsection{Methods}

    \vspace{0.5ex}

\hspace{.8\funcindent}\begin{boxedminipage}{\funcwidth}

    \raggedright \textbf{\_\_init\_\_}(\textit{self})

    \vspace{-1.5ex}

    \rule{\textwidth}{0.5\fboxrule}
\setlength{\parskip}{2ex}
    Initialize the var class

\setlength{\parskip}{1ex}
      \textbf{Parameters}
      \vspace{-1ex}

      \begin{quote}
        \begin{Ventry}{xxxxxxxxxxx}

          \item[type]

          type of mutation ('SNP' or 'INDEL')

            {\it (type=string)}

          \item[pos]

          variant position regarding the reference genome

            {\it (type=integer)}

          \item[var]

          nucleic variation

            {\it (type=string)}

          \item[homoplasy]

          homoplasic variant ('Yes' or 'No')

            {\it (type=string)}

          \item[region]

          type of genetic region ('intergenic' or 'intragenic')

            {\it (type=string)}

          \item[impact]

          variant impact

            {\it (type=string)}

          \item[geneID]

          for a intragenic region, the gene ID impacted by the variant

            {\it (type=string @@param geneName : for a intragenic region, the gene name impacted 
by the variant)}

          \item[transcritID]

          for a intragenic region, the transcrit ID impacted by the variant

            {\it (type=string)}

        \end{Ventry}

      \end{quote}

      Overrides: object.\_\_init\_\_

    \end{boxedminipage}

    \label{script-phyloFixedVar:var:setType}
    \index{script-phyloFixedVar \textit{(script)}!script-phyloFixedVar.var \textit{(class)}!script-phyloFixedVar.var.setType \textit{(method)}}

    \vspace{0.5ex}

\hspace{.8\funcindent}\begin{boxedminipage}{\funcwidth}

    \raggedright \textbf{setType}(\textit{self}, \textit{type})

    \vspace{-1.5ex}

    \rule{\textwidth}{0.5\fboxrule}
\setlength{\parskip}{2ex}
    Set the type of mutation ('SNP' or 'INDEL')

\setlength{\parskip}{1ex}
      \textbf{Parameters}
      \vspace{-1ex}

      \begin{quote}
        \begin{Ventry}{xxxx}

          \item[type]

          type of mutation ('SNP' or 'INDEL')

            {\it (type=string)}

        \end{Ventry}

      \end{quote}

    \end{boxedminipage}

    \label{script-phyloFixedVar:var:setPos}
    \index{script-phyloFixedVar \textit{(script)}!script-phyloFixedVar.var \textit{(class)}!script-phyloFixedVar.var.setPos \textit{(method)}}

    \vspace{0.5ex}

\hspace{.8\funcindent}\begin{boxedminipage}{\funcwidth}

    \raggedright \textbf{setPos}(\textit{self}, \textit{pos})

    \vspace{-1.5ex}

    \rule{\textwidth}{0.5\fboxrule}
\setlength{\parskip}{2ex}
    Set the variant position regarding the reference genome

\setlength{\parskip}{1ex}
      \textbf{Parameters}
      \vspace{-1ex}

      \begin{quote}
        \begin{Ventry}{xxx}

          \item[pos]

          variant position regarding the reference genome

            {\it (type=integer)}

        \end{Ventry}

      \end{quote}

    \end{boxedminipage}

    \label{script-phyloFixedVar:var:setVar}
    \index{script-phyloFixedVar \textit{(script)}!script-phyloFixedVar.var \textit{(class)}!script-phyloFixedVar.var.setVar \textit{(method)}}

    \vspace{0.5ex}

\hspace{.8\funcindent}\begin{boxedminipage}{\funcwidth}

    \raggedright \textbf{setVar}(\textit{self}, \textit{var})

    \vspace{-1.5ex}

    \rule{\textwidth}{0.5\fboxrule}
\setlength{\parskip}{2ex}
    Set the nucleic variation

\setlength{\parskip}{1ex}
      \textbf{Parameters}
      \vspace{-1ex}

      \begin{quote}
        \begin{Ventry}{xxx}

          \item[var]

          nucleic variation

            {\it (type=string)}

        \end{Ventry}

      \end{quote}

    \end{boxedminipage}

    \label{script-phyloFixedVar:var:setHomoplasy}
    \index{script-phyloFixedVar \textit{(script)}!script-phyloFixedVar.var \textit{(class)}!script-phyloFixedVar.var.setHomoplasy \textit{(method)}}

    \vspace{0.5ex}

\hspace{.8\funcindent}\begin{boxedminipage}{\funcwidth}

    \raggedright \textbf{setHomoplasy}(\textit{self}, \textit{homo})

    \vspace{-1.5ex}

    \rule{\textwidth}{0.5\fboxrule}
\setlength{\parskip}{2ex}
    Set if the variant is homoplasic or not ('Yes' or 'No')

\setlength{\parskip}{1ex}
      \textbf{Parameters}
      \vspace{-1ex}

      \begin{quote}
        \begin{Ventry}{xxxx}

          \item[homo]

          homoplasic variant ('Yes' or 'No')

            {\it (type=string)}

        \end{Ventry}

      \end{quote}

    \end{boxedminipage}

    \label{script-phyloFixedVar:var:setRegion}
    \index{script-phyloFixedVar \textit{(script)}!script-phyloFixedVar.var \textit{(class)}!script-phyloFixedVar.var.setRegion \textit{(method)}}

    \vspace{0.5ex}

\hspace{.8\funcindent}\begin{boxedminipage}{\funcwidth}

    \raggedright \textbf{setRegion}(\textit{self}, \textit{reg})

    \vspace{-1.5ex}

    \rule{\textwidth}{0.5\fboxrule}
\setlength{\parskip}{2ex}
    Set the type of genetic region ('intergenic' or 'intragenic')

\setlength{\parskip}{1ex}
      \textbf{Parameters}
      \vspace{-1ex}

      \begin{quote}
        \begin{Ventry}{xxx}

          \item[reg]

          type of genetic region ('intergenic' or 'intragenic')

            {\it (type=string)}

        \end{Ventry}

      \end{quote}

    \end{boxedminipage}

    \label{script-phyloFixedVar:var:setImpact}
    \index{script-phyloFixedVar \textit{(script)}!script-phyloFixedVar.var \textit{(class)}!script-phyloFixedVar.var.setImpact \textit{(method)}}

    \vspace{0.5ex}

\hspace{.8\funcindent}\begin{boxedminipage}{\funcwidth}

    \raggedright \textbf{setImpact}(\textit{self}, \textit{imp})

    \vspace{-1.5ex}

    \rule{\textwidth}{0.5\fboxrule}
\setlength{\parskip}{2ex}
    Set the genetic impact of the variant

\setlength{\parskip}{1ex}
      \textbf{Parameters}
      \vspace{-1ex}

      \begin{quote}
        \begin{Ventry}{xxx}

          \item[imp]

          variant impact

            {\it (type=string)}

        \end{Ventry}

      \end{quote}

    \end{boxedminipage}

    \label{script-phyloFixedVar:var:setGeneID}
    \index{script-phyloFixedVar \textit{(script)}!script-phyloFixedVar.var \textit{(class)}!script-phyloFixedVar.var.setGeneID \textit{(method)}}

    \vspace{0.5ex}

\hspace{.8\funcindent}\begin{boxedminipage}{\funcwidth}

    \raggedright \textbf{setGeneID}(\textit{self}, \textit{id})

    \vspace{-1.5ex}

    \rule{\textwidth}{0.5\fboxrule}
\setlength{\parskip}{2ex}
    Set the gene ID impacted by the variant

\setlength{\parskip}{1ex}
      \textbf{Parameters}
      \vspace{-1ex}

      \begin{quote}
        \begin{Ventry}{xx}

          \item[id]

          gene ID impacted by the variant

            {\it (type=string)}

        \end{Ventry}

      \end{quote}

    \end{boxedminipage}

    \label{script-phyloFixedVar:var:setGeneName}
    \index{script-phyloFixedVar \textit{(script)}!script-phyloFixedVar.var \textit{(class)}!script-phyloFixedVar.var.setGeneName \textit{(method)}}

    \vspace{0.5ex}

\hspace{.8\funcindent}\begin{boxedminipage}{\funcwidth}

    \raggedright \textbf{setGeneName}(\textit{self}, \textit{name})

    \vspace{-1.5ex}

    \rule{\textwidth}{0.5\fboxrule}
\setlength{\parskip}{2ex}
    Set the gene name impacted by the variant

\setlength{\parskip}{1ex}
      \textbf{Parameters}
      \vspace{-1ex}

      \begin{quote}
        \begin{Ventry}{xxxx}

          \item[name]

          gene name impacted by the variant

            {\it (type=string)}

        \end{Ventry}

      \end{quote}

    \end{boxedminipage}

    \label{script-phyloFixedVar:var:setTranscritID}
    \index{script-phyloFixedVar \textit{(script)}!script-phyloFixedVar.var \textit{(class)}!script-phyloFixedVar.var.setTranscritID \textit{(method)}}

    \vspace{0.5ex}

\hspace{.8\funcindent}\begin{boxedminipage}{\funcwidth}

    \raggedright \textbf{setTranscritID}(\textit{self}, \textit{transcritID})

    \vspace{-1.5ex}

    \rule{\textwidth}{0.5\fboxrule}
\setlength{\parskip}{2ex}
    Set the transcrit ID impacted by the variant

\setlength{\parskip}{1ex}
      \textbf{Parameters}
      \vspace{-1ex}

      \begin{quote}
        \begin{Ventry}{xxxxxxxxxxx}

          \item[transcritID]

          transcrit ID impacted by the variant

            {\it (type=string)}

        \end{Ventry}

      \end{quote}

    \end{boxedminipage}


\large{\textbf{\textit{Inherited from object}}}

\begin{quote}
\_\_delattr\_\_(), \_\_format\_\_(), \_\_getattribute\_\_(), \_\_hash\_\_(), \_\_new\_\_(), \_\_reduce\_\_(), \_\_reduce\_ex\_\_(), \_\_repr\_\_(), \_\_setattr\_\_(), \_\_sizeof\_\_(), \_\_str\_\_(), \_\_subclasshook\_\_()
\end{quote}

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
%%                              Properties                               %%
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

  \subsubsection{Properties}

    \vspace{-1cm}
\hspace{\varindent}\begin{longtable}{|p{\varnamewidth}|p{\vardescrwidth}|l}
\cline{1-2}
\cline{1-2} \centering \textbf{Name} & \centering \textbf{Description}& \\
\cline{1-2}
\endhead\cline{1-2}\multicolumn{3}{r}{\small\textit{continued on next page}}\\\endfoot\cline{1-2}
\endlastfoot\multicolumn{2}{|l|}{\textit{Inherited from object}}\\
\multicolumn{2}{|p{\varwidth}|}{\raggedright \_\_class\_\_}\\
\cline{1-2}
\end{longtable}

    \index{script-phyloFixedVar \textit{(script)}!script-phyloFixedVar.var \textit{(class)}|)}

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
%%                           Class Description                           %%
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

    \index{script-phyloFixedVar \textit{(script)}!script-phyloFixedVar.genome \textit{(class)}|(}
\subsection{Class genome}

    \label{script-phyloFixedVar:genome}
\begin{tabular}{cccccc}
% Line for object, linespec=[False]
\multicolumn{2}{r}{\settowidth{\BCL}{object}\multirow{2}{\BCL}{object}}
&&
  \\\cline{3-3}
  &&\multicolumn{1}{c|}{}
&&
  \\
&&\multicolumn{2}{l}{\textbf{script-phyloFixedVar.genome}}
\end{tabular}

Create genome object to stock all of its variants informations


%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
%%                                Methods                                %%
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

  \subsubsection{Methods}

    \vspace{0.5ex}

\hspace{.8\funcindent}\begin{boxedminipage}{\funcwidth}

    \raggedright \textbf{\_\_init\_\_}(\textit{self})

    \vspace{-1.5ex}

    \rule{\textwidth}{0.5\fboxrule}
\setlength{\parskip}{2ex}
    Initialize the genome class

\setlength{\parskip}{1ex}
      \textbf{Parameters}
      \vspace{-1ex}

      \begin{quote}
        \begin{Ventry}{xxxxxxxx}

          \item[name]

          name of the genome

            {\it (type=string)}

          \item[variants]

          list of var objects assigned to the genome

            {\it (type=list)}

        \end{Ventry}

      \end{quote}

      Overrides: object.\_\_init\_\_

    \end{boxedminipage}

    \label{script-phyloFixedVar:genome:setName}
    \index{script-phyloFixedVar \textit{(script)}!script-phyloFixedVar.genome \textit{(class)}!script-phyloFixedVar.genome.setName \textit{(method)}}

    \vspace{0.5ex}

\hspace{.8\funcindent}\begin{boxedminipage}{\funcwidth}

    \raggedright \textbf{setName}(\textit{self}, \textit{name})

    \vspace{-1.5ex}

    \rule{\textwidth}{0.5\fboxrule}
\setlength{\parskip}{2ex}
    Set the name of the genome

\setlength{\parskip}{1ex}
      \textbf{Parameters}
      \vspace{-1ex}

      \begin{quote}
        \begin{Ventry}{xxxx}

          \item[name]

          name of the genome

            {\it (type=string)}

        \end{Ventry}

      \end{quote}

    \end{boxedminipage}

    \label{script-phyloFixedVar:genome:setVar}
    \index{script-phyloFixedVar \textit{(script)}!script-phyloFixedVar.genome \textit{(class)}!script-phyloFixedVar.genome.setVar \textit{(method)}}

    \vspace{0.5ex}

\hspace{.8\funcindent}\begin{boxedminipage}{\funcwidth}

    \raggedright \textbf{setVar}(\textit{self}, \textit{variants})

    \vspace{-1.5ex}

    \rule{\textwidth}{0.5\fboxrule}
\setlength{\parskip}{2ex}
    Set a list of var objects assigned to the genome

\setlength{\parskip}{1ex}
      \textbf{Parameters}
      \vspace{-1ex}

      \begin{quote}
        \begin{Ventry}{xxxxxxxx}

          \item[variants]

          list of var objects assigned to the genome

            {\it (type=list)}

        \end{Ventry}

      \end{quote}

    \end{boxedminipage}

    \label{script-phyloFixedVar:genome:addVariant}
    \index{script-phyloFixedVar \textit{(script)}!script-phyloFixedVar.genome \textit{(class)}!script-phyloFixedVar.genome.addVariant \textit{(method)}}

    \vspace{0.5ex}

\hspace{.8\funcindent}\begin{boxedminipage}{\funcwidth}

    \raggedright \textbf{addVariant}(\textit{self}, \textit{variant})

    \vspace{-1.5ex}

    \rule{\textwidth}{0.5\fboxrule}
\setlength{\parskip}{2ex}
    Add a new var object in the genome variants list

\setlength{\parskip}{1ex}
      \textbf{Parameters}
      \vspace{-1ex}

      \begin{quote}
        \begin{Ventry}{xxxxxxxx}

          \item[variants]

          list of var objects assigned to the genome

            {\it (type=list)}

        \end{Ventry}

      \end{quote}

    \end{boxedminipage}


\large{\textbf{\textit{Inherited from object}}}

\begin{quote}
\_\_delattr\_\_(), \_\_format\_\_(), \_\_getattribute\_\_(), \_\_hash\_\_(), \_\_new\_\_(), \_\_reduce\_\_(), \_\_reduce\_ex\_\_(), \_\_repr\_\_(), \_\_setattr\_\_(), \_\_sizeof\_\_(), \_\_str\_\_(), \_\_subclasshook\_\_()
\end{quote}

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
%%                              Properties                               %%
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

  \subsubsection{Properties}

    \vspace{-1cm}
\hspace{\varindent}\begin{longtable}{|p{\varnamewidth}|p{\vardescrwidth}|l}
\cline{1-2}
\cline{1-2} \centering \textbf{Name} & \centering \textbf{Description}& \\
\cline{1-2}
\endhead\cline{1-2}\multicolumn{3}{r}{\small\textit{continued on next page}}\\\endfoot\cline{1-2}
\endlastfoot\multicolumn{2}{|l|}{\textit{Inherited from object}}\\
\multicolumn{2}{|p{\varwidth}|}{\raggedright \_\_class\_\_}\\
\cline{1-2}
\end{longtable}

    \index{script-phyloFixedVar \textit{(script)}!script-phyloFixedVar.genome \textit{(class)}|)}

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
%%                           Class Description                           %%
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

    \index{script-phyloFixedVar \textit{(script)}!script-phyloFixedVar.node \textit{(class)}|(}
\subsection{Class node}

    \label{script-phyloFixedVar:node}
\begin{tabular}{cccccc}
% Line for object, linespec=[False]
\multicolumn{2}{r}{\settowidth{\BCL}{object}\multirow{2}{\BCL}{object}}
&&
  \\\cline{3-3}
  &&\multicolumn{1}{c|}{}
&&
  \\
&&\multicolumn{2}{l}{\textbf{script-phyloFixedVar.node}}
\end{tabular}

Create node object to stock the phylogeny tree


%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
%%                                Methods                                %%
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

  \subsubsection{Methods}

    \vspace{0.5ex}

\hspace{.8\funcindent}\begin{boxedminipage}{\funcwidth}

    \raggedright \textbf{\_\_init\_\_}(\textit{self})

    \vspace{-1.5ex}

    \rule{\textwidth}{0.5\fboxrule}
\setlength{\parskip}{2ex}
    Initialize the node class

\setlength{\parskip}{1ex}
      \textbf{Parameters}
      \vspace{-1ex}

      \begin{quote}
        \begin{Ventry}{xxxxxxxx}

          \item[fils]

          list of sons nodes objects

            {\it (type=list)}

          \item[pere]

          father node object

            {\it (type=node object)}

          \item[val]

          genome name if the node is a leaf

            {\it (type=string)}

          \item[name]

          name of the genome

            {\it (type=string)}

          \item[variants]

          list of var objects assigned to the genome

            {\it (type=list)}

        \end{Ventry}

      \end{quote}

      Overrides: object.\_\_init\_\_

    \end{boxedminipage}

    \label{script-phyloFixedVar:node:nbFils}
    \index{script-phyloFixedVar \textit{(script)}!script-phyloFixedVar.node \textit{(class)}!script-phyloFixedVar.node.nbFils \textit{(method)}}

    \vspace{0.5ex}

\hspace{.8\funcindent}\begin{boxedminipage}{\funcwidth}

    \raggedright \textbf{nbFils}(\textit{self})

    \vspace{-1.5ex}

    \rule{\textwidth}{0.5\fboxrule}
\setlength{\parskip}{2ex}
    Compute number of sons

\setlength{\parskip}{1ex}
      \textbf{Return Value}
    \vspace{-1ex}

      \begin{quote}
      number of sons

      {\it (type=integer)}

      \end{quote}

    \end{boxedminipage}

    \label{script-phyloFixedVar:node:isLeaf}
    \index{script-phyloFixedVar \textit{(script)}!script-phyloFixedVar.node \textit{(class)}!script-phyloFixedVar.node.isLeaf \textit{(method)}}

    \vspace{0.5ex}

\hspace{.8\funcindent}\begin{boxedminipage}{\funcwidth}

    \raggedright \textbf{isLeaf}(\textit{self})

    \vspace{-1.5ex}

    \rule{\textwidth}{0.5\fboxrule}
\setlength{\parskip}{2ex}
    Check if the node is a leaf

\setlength{\parskip}{1ex}
      \textbf{Return Value}
    \vspace{-1ex}

      \begin{quote}
      True if the node is a leaf, False otherwise

      {\it (type=boolean)}

      \end{quote}

    \end{boxedminipage}

    \label{script-phyloFixedVar:node:setPere}
    \index{script-phyloFixedVar \textit{(script)}!script-phyloFixedVar.node \textit{(class)}!script-phyloFixedVar.node.setPere \textit{(method)}}

    \vspace{0.5ex}

\hspace{.8\funcindent}\begin{boxedminipage}{\funcwidth}

    \raggedright \textbf{setPere}(\textit{self}, \textit{node})

    \vspace{-1.5ex}

    \rule{\textwidth}{0.5\fboxrule}
\setlength{\parskip}{2ex}
    Set the father node of the current node object

\setlength{\parskip}{1ex}
      \textbf{Parameters}
      \vspace{-1ex}

      \begin{quote}
        \begin{Ventry}{xxxx}

          \item[node]

          father node object

            {\it (type=node object)}

        \end{Ventry}

      \end{quote}

    \end{boxedminipage}

    \label{script-phyloFixedVar:node:setFils}
    \index{script-phyloFixedVar \textit{(script)}!script-phyloFixedVar.node \textit{(class)}!script-phyloFixedVar.node.setFils \textit{(method)}}

    \vspace{0.5ex}

\hspace{.8\funcindent}\begin{boxedminipage}{\funcwidth}

    \raggedright \textbf{setFils}(\textit{self}, \textit{node})

    \vspace{-1.5ex}

    \rule{\textwidth}{0.5\fboxrule}
\setlength{\parskip}{2ex}
    Add a new son to the current node object

\setlength{\parskip}{1ex}
      \textbf{Parameters}
      \vspace{-1ex}

      \begin{quote}
        \begin{Ventry}{xxxx}

          \item[node]

          son node object

            {\it (type=node object)}

        \end{Ventry}

      \end{quote}

    \end{boxedminipage}

    \label{script-phyloFixedVar:node:setVal}
    \index{script-phyloFixedVar \textit{(script)}!script-phyloFixedVar.node \textit{(class)}!script-phyloFixedVar.node.setVal \textit{(method)}}

    \vspace{0.5ex}

\hspace{.8\funcindent}\begin{boxedminipage}{\funcwidth}

    \raggedright \textbf{setVal}(\textit{self}, \textit{val})

    \vspace{-1.5ex}

    \rule{\textwidth}{0.5\fboxrule}
\setlength{\parskip}{2ex}
    Set the genome name of the node

\setlength{\parskip}{1ex}
      \textbf{Parameters}
      \vspace{-1ex}

      \begin{quote}
        \begin{Ventry}{xxx}

          \item[val]

          genome name

            {\it (type=string)}

        \end{Ventry}

      \end{quote}

    \end{boxedminipage}

    \label{script-phyloFixedVar:node:listLeaf}
    \index{script-phyloFixedVar \textit{(script)}!script-phyloFixedVar.node \textit{(class)}!script-phyloFixedVar.node.listLeaf \textit{(method)}}

    \vspace{0.5ex}

\hspace{.8\funcindent}\begin{boxedminipage}{\funcwidth}

    \raggedright \textbf{listLeaf}(\textit{self}, \textit{liste})

    \vspace{-1.5ex}

    \rule{\textwidth}{0.5\fboxrule}
\setlength{\parskip}{2ex}
    Give all genomes name of all sons node objects (recursive function)

\setlength{\parskip}{1ex}
      \textbf{Parameters}
      \vspace{-1ex}

      \begin{quote}
        \begin{Ventry}{xxxxx}

          \item[liste]

          empty list which will contain genomes name

            {\it (type=list)}

        \end{Ventry}

      \end{quote}

    \end{boxedminipage}


\large{\textbf{\textit{Inherited from object}}}

\begin{quote}
\_\_delattr\_\_(), \_\_format\_\_(), \_\_getattribute\_\_(), \_\_hash\_\_(), \_\_new\_\_(), \_\_reduce\_\_(), \_\_reduce\_ex\_\_(), \_\_repr\_\_(), \_\_setattr\_\_(), \_\_sizeof\_\_(), \_\_str\_\_(), \_\_subclasshook\_\_()
\end{quote}

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
%%                              Properties                               %%
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

  \subsubsection{Properties}

    \vspace{-1cm}
\hspace{\varindent}\begin{longtable}{|p{\varnamewidth}|p{\vardescrwidth}|l}
\cline{1-2}
\cline{1-2} \centering \textbf{Name} & \centering \textbf{Description}& \\
\cline{1-2}
\endhead\cline{1-2}\multicolumn{3}{r}{\small\textit{continued on next page}}\\\endfoot\cline{1-2}
\endlastfoot\multicolumn{2}{|l|}{\textit{Inherited from object}}\\
\multicolumn{2}{|p{\varwidth}|}{\raggedright \_\_class\_\_}\\
\cline{1-2}
\end{longtable}

    \index{script-phyloFixedVar \textit{(script)}!script-phyloFixedVar.node \textit{(class)}|)}
    \index{script-phyloFixedVar \textit{(script)}|)}
